src/ui/SurveyUnit/Communication/CommunicationForm.tsx

```diff
@@ -0,0 +1,218 @@
+  const [step, setStep] = useState(Steps.MEDIUM);
+  const [communicationRequest, setCommunicationRequest] = useState({
+    medium: '',
+    reason: '',
+    type: '',
+  } as CommunicationRequestForm);
+
+  const setCommunicationRequestPropValue = (value: string) => {
+    const prop = Steps[step].toLocaleLowerCase();
+    setCommunicationRequest({ ...communicationRequest, [prop]: value });
+  };
+
+  let communicationTemplates = surveyUnit?.communicationTemplates ?? [];
+  const mediums = mediumRadioValues.map(m => {
+    return {
+      value: m.value,
+      label: m.label,
+      disabled: !communicationTemplates.some(c => c.medium === m.value),
+    } as CommunicationRequestRadioData;
+  });
+
+  // Will allow the user to pick only the types found in communicationTemplates with the same medium as the one picked by the user (communicationRequest.medium)
+  communicationTemplates = communicationTemplates.filter(
+    item => item.medium === communicationRequest.medium
+  );
+
+  const types = typeRadioValues.map(t => {
+    return {
+      value: t.value,
+      label: t.label,
+      disabled: !communicationTemplates.some(c => c.type === t.value),
+    } as CommunicationRequestRadioData;
+  });
+
+  const bypassTypeValue = communicationTypeEnum.COMMUNICATION_NOTICE.value;
+  const bypassReasonValue = communicationReasonEnum.UNREACHABLE.value;
+
+  const reasons = reasonRadioValues.map(r => {
+    return {
+      value: r.value,
+      label: r.label,
+      disabled: !communicationTemplates.some(c => c.type !== bypassTypeValue),
+    } as CommunicationRequestRadioData;
+  });
+
+  const saveCommunicationRequest = () => {
+    // Retrieveing communicationTemplatedId by using form's input from the user
+    const communicationTemplatedId = communicationTemplates.find(
+      communicationTemplate =>
+        communicationTemplate.type === communicationRequest.type &&
+        communicationTemplate.medium === communicationRequest.medium
+    )?.id;
+
+    const newCommunicationRequest = {
+      reason: communicationRequest.reason,
+      emitter: 'INTERVIEWER',
+      communicationTemplateId: communicationTemplatedId,
+      status: [{ date: new Date().getTime(), status: communicationStatusEnum.INITIATED.value }],
+    } as SurveyUnitCommunicationRequest;
+
+    surveyUnitIDBService.addOrUpdateSU({
+      ...surveyUnit,
+      communicationRequests: [...(surveyUnit.communicationRequests ?? []), newCommunicationRequest],
+    });
+
+    onClose();
+  };
+
+  const nextStep = () => {
+    if (step < Steps.VALIDATE) setStep(Steps.after(step));
+  };
+
+  const bypass = communicationRequest.type === bypassTypeValue && step === Steps.REASON;
+  const bypassed = communicationRequest.type === bypassTypeValue && step === Steps.VALIDATE;
+
+  const previousStep = () => {
+    if (bypassed) {
+      setStep(Steps.before(Steps.before(step)));
+    } else if (step !== Steps.MEDIUM) {
+      setStep(Steps.before(step));
+    }
+  };
+
+  useMemo(() => {
```

Comment:
A `useMemo` shoudl always be used with an assignment : `const data = useMemo(...)`

In your case I think you need a useEffect maybe ?